APCAppCore/APCAppCore/Library/FeedParser/APCFeedParser.m

```diff
@@ -0,0 +1,259 @@
+        } else if ([elementName isEqualToString:@"content:encoded"] || [elementName isEqualToString:@"content"]) {
+            self.feedItem.content = self.parsedString;
+        } else if ([elementName isEqualToString:@"link"]) {
+            self.feedItem.link = self.parsedString;
+        } else if ([elementName isEqualToString:@"pubDate"]) {
+            _dateFormatter.dateFormat = @"EEE, dd MMM yyyy HH:mm:ss Z";
+            self.feedItem.pubDate = [self.dateFormatter dateFromString:self.parsedString];
+        } else if ([elementName isEqualToString:@"dc:creator"]) {
+            self.feedItem.author = self.parsedString;
+        } else if ([elementName isEqualToString:@"guid"]) {
+            self.feedItem.guid = self.parsedString;
+        }
+        
+        // sometimes the URL is inside enclosure element, not in link. Reference: http://www.w3schools.com/rss/rss_tag_enclosure.asp
+        if ([elementName isEqualToString:@"enclosure"] && self.attributeDict != nil) {
+            NSString *url = [self.attributeDict objectForKey:@"url"];
+            if(url) {
+                self.feedItem.link = url;
+            }
+        }
+    }
+}
+
+- (void)parserDidEndDocument:(NSXMLParser *)__unused parser
+{
+    __weak typeof(self) weakSelf = self;
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (weakSelf.completionBlock) {
+            weakSelf.completionBlock([NSArray arrayWithArray:self.results], nil);
+        }
+    });
+}
+
+- (void)parser:(NSXMLParser *)parser parseErrorOccurred:(NSError *)parseError
+{
+    __weak typeof(self) weakSelf = self;
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (weakSelf.completionBlock) {
+            weakSelf.completionBlock(nil, parseError);
+        }
+    });
+    
+    [parser abortParsing];
+}
+
+- (void)parser:(NSXMLParser *)parser validationErrorOccurred:(NSError *)validationError
+{
+    __weak typeof(self) weakSelf = self;
+    
+    dispatch_async(dispatch_get_main_queue(), ^{
+        if (weakSelf.completionBlock) {
+            weakSelf.completionBlock(nil, validationError);
+        }
+    });
+    
+    [parser abortParsing];
+}
+                         
+@end
+
+
+@implementation APCFeedItem
+
+- (instancetype)init
+{
+    self = [super init];
+    if (self) {
+        _title = @"";
+        _link = @"";
+        _content = @"";
+        _itemDescription = @"";
+        
+    }
+    return self;
+}
+
+- (NSArray *)imagesFromItemDescription
+{
+    NSArray *images = nil;
+    
+    if (self.itemDescription) {
+        images = [self imagesFromHTMLString:self.itemDescription];
+    }
+    
+    return images;
+}
+
+- (NSArray *)imagesFromContent
+{
+    NSArray *images = nil;
+    
+    if (self.content) {
+        images = [self imagesFromHTMLString:self.content];
+    }
+    
+    return images;
+}
+
+#pragma mark - retrieve images from html string using regexp
+
+- (NSArray *)imagesFromHTMLString:(NSString *)htmlstr
+{
```

Comment:
Better to rename this method to `imageURLsFromHTMLString`?

`imageURLsFromContent` 
`imagesURLsFromItemDescription `